utils/refchecker.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import argparse
+import json
+
+import ssg.build_yaml
+import ssg.products
+import ssg.rules
+import ssg.yaml
+import ssg.utils
+import ssg.rule_yaml
+
+SSG_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-j", "--json", type=str, action="store",
+                        default="build/rule_dirs.json", help="File to read "
+                        "json output of rule_dir_json from (defaults to "
+                        "build/rule_dirs.json")
+    parser.add_argument("-c", "--build-config-yaml", default="build/build_config.yml",
+                        help="YAML file with information about the build configuration. "
+                        "Defaults to build/build_config.yml")
+    parser.add_argument("-p", "--profiles-root",
+                        help="Override where to look for profile files.")
+    parser.add_argument("product", type=str, help="Product to check has required references")
+    parser.add_argument("profile", type=str, help="Profile to iterate over")
+    parser.add_argument("reference", type=str, help="Required reference system to check for")
+
+    return parser.parse_args()
+
+
+def load(rule_obj, env_yaml=None, sce_metadata=None):
+    """
+    From the given rule_dir object, load the build_yaml.Rule associated with
+    it.
+    """
+
+    yaml_file = ssg.rules.get_rule_dir_yaml(rule_obj['dir'])
+    return ssg.build_yaml.Rule.from_yaml(yaml_file, env_yaml=env_yaml,
+                                         sce_metadata=sce_metadata)
```

Comment:
The sce_metadata doesn't exist

Reply:
Sorry, that's an unrelated change that crept in to this utility. :)

Reply:
Yes, but because of this variable it doesn't work.

```
Traceback (most recent call last):
  File "/home/jcerny/work/git/scap-security-guide/utils/refchecker.py", line 114, in <module>
    main()
  File "/home/jcerny/work/git/scap-security-guide/utils/refchecker.py", line 107, in main
    ok = reference_check(env_yaml, all_rules, profile_path, args.product, args.reference)
  File "/home/jcerny/work/git/scap-security-guide/utils/refchecker.py", line 68, in reference_check
    rule = load_for_product(rule_dirs[rule_id], product, env_yaml=env_yaml)
  File "/home/jcerny/work/git/scap-security-guide/utils/refchecker.py", line 53, in load_for_product
    rule = load(rule_obj, env_yaml=env_yaml, sce_metadata=sce_metadata)
  File "/home/jcerny/work/git/scap-security-guide/utils/refchecker.py", line 43, in load
    return ssg.build_yaml.Rule.from_yaml(yaml_file, env_yaml=env_yaml,
TypeError: from_yaml() got an unexpected keyword argument 'sce_metadata'
```